### DISCUSSION TITLE
[Balance] Unlimited Oxygen and Batteries

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#9669] [Ideas and suggestions]
So here's my rant about Oxygen and Batteries (updated)

# Oxygen

Currently, you can stack 8 O2 bottles in a single inventory slot.
One oxygen bottle can last 5 minutes and 33 seconds on a regular diving suit
![image](https://user-images.githubusercontent.com/73229309/182129763-e43cd6cd-e613-4f2a-8c40-a91ce617b117.png)

**With 8 bottles in inventory and 1 in the suit, you can have up to 45 MINUTES of oxygen**
This is before condition is even factored; masterwork increases that duration by 30%. DOUBLE that if you're wearing an Abyss suit.
It's not uncommon for crewmates to wear diving suits the entire round that way

_It is virtually impossible to suffocate in this underwater game._

You can find suits and oxygen bottles in pristine condition inside any wrecked cabinet and beacons, allowing you to swap oxygen bottles for another 5+ minutes of air (more if you find quality bottles):
![image](https://user-images.githubusercontent.com/73229309/182132297-4722bc65-6794-489a-b2a0-613c648b21f0.png)
![image](https://user-images.githubusercontent.com/73229309/182132550-a73a940d-3d8d-4faf-ae1f-75c4543335b7.png)

As a result, I regularly toss away empty bottles and replace them with fresh ones inside wrecks. 

Condition and Quality also doesn't stack, which means bringing a stack of several bottles will eventually split into separate slots.
![image](https://user-images.githubusercontent.com/73229309/182137496-1de59933-ad33-4960-96b1-de386cbb9378.png)

If your stack of O2 gets damaged by a fire, all of the bottles lose their stacking ability, and you cannot combine them until they've been taken out of the stack one by one
![image](https://user-images.githubusercontent.com/73229309/210607189-54014bf9-9f92-4af3-b069-f11f88b091ae.png)

You can leave diving suits all over the ship on the floor with no downside, pick them up whenever there's a flood and be safe from pressure and low oxygen
![image](https://user-images.githubusercontent.com/73229309/210613470-d7257503-b10e-4394-af95-6fbecf8fd3d2.png)

Overall there's too much oxygen:
-**HIt oxygen tanks with the biggest nerfbat there is and limit the amount of oxygen bottles to 1 per slot**
-Make the suits "leak" oxygen when not stored in their cabinets
-Reduce the quantity/condition of oxygen bottles found in wrecks and beacon cabinets to 0 _(turns out there's currently no existing parameter to define spawn quality inside wrecks)_
-Increase Abyss Suit O2 consumption (it's already the sturdiest suit)

```xml
		<!-- Leaky suit when not contained -->
      <StatusEffect type="OnNotContained" target="Contained,NearbyCharacters" OxygenAvailable="1000.0" Condition="-1" limb="Head">
        <RequiredItem items="oxygentank,weldingfueltank,incendiumfueltank,oxygenitetank" type="Contained" />
        <Sound file="Content/Items/Diving/ITEMS_divingSuitOxygenLeakLoop.ogg" range="300" volume="0.5" loop="true" />
		<ParticleEmitter particle="spray" anglemin="45" anglemax="135" velocitymin="5.0" velocitymax="10.0" particlespersecond="10" scalemin="0.2" scalemax="0.3" colormultiplier="240,240,100,255" />
		<ParticleEmitter particle="bubbles" anglemin="45" anglemax="135" velocitymin="5.0" velocitymax="10.0" particlespersecond="2" scalemin="0.2" scalemax="0.3" colormultiplier="240,240,100,255" />
		<Conditional condition="gt 1.0" />
      </StatusEffect>	
```

# Batteries

The only two items where batteries are limiting factor are Stun Batons and Portable Pumps, which only have a few hits/seconds of reserve.
**UPDATE:
Many more items that use batteries have been introduced since, so battery quality and quantity does matter more**

Battery duration for other items are:
-Flashlight: 500 seconds (8 minutes 20 seconds)
-Scooter: 400 seconds (6 minutes 40 seconds)
-Sonar beacon: 1000 seconds (over 16 minutes and it attacks monsters like a decoy)
-Handheld sonar: 100 seconds (but you'll be turning that off when not looking at it)

_It is virtually impossible to run out of power for the items above._

This is ONE battery of normal quality and you can stack them up to 8x.
Fulgurium battery cells DOUBLE that duration, and unlike oxygenite tanks, they are refillable!

Worse, battery power can't be combined/transfered, cluttering the inventory if you have partially used batteries.

Swing the nerfbat-tery:
-Consider removing stacking

**Comment by MarkCringan** [comment]
I feel like there's an interesting idea to be had here. What about something that behaves like a disposable diving suit and can only be refilled by suit docks. Maybe give it 20ish minutes of air and no refilling outside the sub. Adding an item like this could allow more hard-core players to use it whilst still allowing people who like the current system to use it.

Recipie could be combining 2 oxygen tanks (or so) with a suit of your choice to make the special version of that suit. Idk, just throwing out ideas.

Also, you could make a version of electrical devices in a similar way with the battery built in too! Lasts longer but can only be refilled on the sub.

I'd like to hear what y'all think about this, as I also (kinda) feel like having a suit on all the time is a bit cheezy.

**Reply by Ek-Videogames** [reply]
In effect, one oxygen tank that just lasts a _reasonable_ amount of time - solving the reason why you always _had_ to bring 8 tanks to begin with.

**Reply by ProphetCrow** [reply]
It makes forest caves more dangerous too. If I remember correctly, the gas vents drain by % currently. This means with this idea, You can lose 20m of air in 100 seconds. Tweak the numbers or variables, could be fine. Id honestly would prefer a system of larger, nonstackable oxycans that each last a decent amount of time. But if you do it like that, you may also want to rework welding fuel canisters due to similar sprite design and slot compatibility.

Also, consider how diving masks would interact with this system. 

If Oxy tanks were made unstackable and the capacity increased, and crafting reqs and price of the tanks should also increase while decreasing the overall availability/quantity, obviously. 

This would make it so you'd have to trade more inventory space for longer EVA time, or actually risk suffocation.

**Comment by NonCancerAlias** [comment]
You make a decent argument but I for one think this is a terrible idea, simply put if something isn’t broken don’t try to fix it.

Furthermore oxygen tanks aren’t supposed to be some giant issue to manage and plan everything around, in my opinion it’s a secondary environmental issue that only really strikes after it’s been neglected in favor of way more pressing issues

**Comment by ubertpendragon** [comment]
There seems to be a misconception about what would happen to EVA autonomy without O2 stacking, so here's a simple practical example:

A single O2 bottle lasts 333 seconds. 9 bottles (1 in the suit + 8x stack) last 50 minutes. 

**By removing stacking, this would be cut down to 11 minutes, on the same inventory space.**
16 minutes if you sacrifice 1 extra slot, 22 if you sacrifice 2... This is very generous

Factor in:
-Quality, a mechanic with base stats can increase the duration of a bottle to 6 minutes
-Oxygenite tanks which significantly increase diving duration
-Abyss suits reducing O2 consumption
-Extra inventory slots from toolbelts, cargo scooters, exosuits...
-Tiger thresher genes reducing O2 consumption

Taking away the stacking (and other suggested) merely introduces a choice/risk of running out of O2, changes _which currently does not happen_ when every sub has a suit on the floor and every crewmate 8 bottles in their inventory


And yes I know about the green gas vents, you can nerf those.

**Reply by ProphetCrow** [reply]
Yeah thats cool, I'm on the same page. Only change Id suggest is make o2 tanks maybe 500 seconds instead, something to compensate players to what would be a sudden change.

**Comment by Bebrich** [comment]
I'm not sure about the removing the ability to stack tanks, as this could make the game more tedious, but I've made a proposal to fix the stacking problem in #11368

**Comment by Fentanyl1984** [comment]
My suggestions for oxygen tanks nerf:
- outright remove oxy tanks qualities. Like even from logical standpoint, do they get bigger with every quality level or smtg??doesnt make sense
- imo removing oxy tanks stacks is too harsh of a nerf. reducing it down to 3-4 is enough
Aaaand thats it i think. More nerfs would only make oxy managament more annoying

**Reply by Bebrich** [reply]
Removing qualities as a balance change doesn't make sense

**Reply by Saevon** [reply]
Quality can make sense just fine,  "its built better so has better pressure"

Stacks down to 3 is good! that means one stack (+suit) is 22min

**Comment by BiffW** [comment]
Removing the stacking ability of oxygen tanks can cause backwards compatibility issue because all the saves in current version have oxygen tanks stacking together.

**Comment by BiffW** [comment]
Instead of a direct nerf to oxygen tank. I think it is a better idea to create a new threat. For instance, a new enemy that shoots acid spikes which boost the victim's oxygen consumption rate.
And I think the oxygen consumption logic needs a total rework. Currently it is defined in diving suits and has no relation to the characters themselves. The tiger thresher gene doesn't really reduce the rate you consume oxygen but provide oxygen low resistance which means you consume oxygen at exactly the same rate as not having the gene.

**Comment by aeondez** [comment]
I am in agreement with some but not all of this. TL;DR: Batteries are fine as-is, but Fulgurium should be single use. O2 is overly generous, tank quality should be removed, and stack size should be reduced to x1 or x2.

General thoughts on batteries:
I like the idea of battery quality. It makes sense, and feels fun to be able to craft. The main utility uses for batteries are generally going to be scooters and handheld sonars, and a standard battery is almost always enough for normal usage, as long as they are recharged after each EVA. Miniature piezo crystals in caves that zap batteries would be hilarious and feel dangerous because you may get stranded or lost, but that's a different sort of idea.

Duration/Condition: 
Batteries generally hold a reasonable amount of charge. I don't think they really need any changes.

Stack size: 
Batteries could use a stack size nerf, but something like x2 or x4. One slot per battery is too restrictive when carrying spares for things like arc emitters, stun batons, or Harpoon Coil-Rifles.

Quality: 
Battery quality also makes sense, and I see no real issues with it at present.

Fulgurium batteries: 
I wouldn't care one way or the other if Fulgurium batteries were one-time use or not, but do lean towards them being single-use batteries. It would make sense to have Fulgurium batteries to be one-time use items to bring them in line with Incendium tanks and Oxygenite tanks, but they may need to have their charge increased a bit to make them worth crafting instead of Fulgurium fuel rods.


General thoughts on Oxygen: 
Oxygen definitely needs a balance pass. Running out of O2 feels like an annoyance rather than a threat. I don't care for the idea of leaky suits, but I would like to see bots put them away after using them without being prompted. Also, O2 generators on vanilla subs are overly generous and should be adjusted a bit. 

Duration/Condition: 
I don't think the duration needs to be changed as long as stack size is nerfed heavily. The current method of combining partial O2 tanks to get empty/full tanks is fine by me. As far as condition damage to an entire stack (such as your example with fire) goes, that's a niche use case for me, and reducing stack size to x1 or x2 would probably make that even more of a niche issue for most.

Stack size: 
One or two per stack makes sense. If you have one in a suit, plus a spare stack of 1 or 2, plus an emergency tank in your plasma cutter, that's still 15-20 minutes of O2, which feels just right for EVA or major engagements with Abyssal creatures. Prioritizing extra O2 for ruins or caves means leaving empty tanks behind to get more loot.

Quality: 
Oxygen tank quality is massively annoying to organize on a sub, and makes finding wreck oxygen tanks basically worthless except as a source of aluminum. It doesn't make sense to have higher quality tanks and I'd like to see them go away.

Oxygenite tanks:
No issues, works fine. Oxygenite is versatile, and oxygenite tanks are a solid choice for using the oxygenite shards.

**Comment by Ek-Videogames** [comment]
It's kind of funny how this has gone full circle, oxygen tanks got nerfed (obviously) because of the addition of stacking - and now people are complaining that players are carrying around too much oxygen tanks.

As for batteries, some items use way more battery power than others (again, thus requiring people carrying full stacks of them because 1 spare battery simply isn't enough). This problem existed as early as the pre-steam beta, when stun batons only had a measly TWO swings before being functionally worthless, leading to security officers hording batteries.

**Comment by duarm** [comment]
At the very least, suits should leak oxygen when on ground, this would be a step towards a fun optimal gameplay for barotrauma, currently playing barotrauma optimally breaks the atmosphere and makes most jobs idle (reactor circuit,  emergency lamp rewiring, suits dropped everywhere, etc.).

**Reply by adamkad1** [reply]
I allways try to keep a submarine tidy, but unfortunately, bots do what players do and just drop a suit when they are done with it.

**Comment by adamkad1** [comment]
Honestly, I wouldnt mind removing stacking, anyone bothered with it can just download a backpack mod to have all the space theyll ever need for cargo and oxygen tanks

**Comment by 8on8er** [comment]
While I agree with a lot of points you have made in your post I also agree with some of the good points made in the replies.

Although the total amount of time you can spend in the water due to oxygen tanks when you take an objective look at it is rather obscene and poses no real risk to players. Like others I feel like it is more of a secondary negligent risk which can often happen if you forget to take enough tanks with you. Furthermore, I feel like nerfing oxygen tanks on the line of fun vs challenging gameplay is too close to challenging. I feel most players rather the challenge of the game be fighting monsters rather than counting oxygen tanks.

I feel that as a mod alongside things like neurotrauma or realistic sonar making oxygen management more punishing could be more fun in it's own horror way but I don't think it belongs in the base game.

**Comment by Arctodites** [comment]
Yes, please nerf O2 tanks!  I don't like the idea of removing oxygen tank stackability, but making suits slowly 'leak' when they're not being worn or stored in a suit locker is a brilliant idea.  It would discourage players from nicking the diving suit nearest my station on the sub and not returning it for the whole round (or ever) just because it happened to be the closest one they could find.  It could also make diving masks more distinctively useful, because right now most players just wear a diving suit everywhere and treat it as body armour.

A few other ways that O2 could be nerfed: obviously getting rid of their item quality levels.  But also reducing the capacity of oxygen refill shelves from 3 to 2 (as suggested by IgRAzm a few weeks ago: [#13036](https://github.com/Regalis11/Barotrauma/discussions/13036#discussioncomment-7753640).  Another would be to only allow certain, higher tier suits to be compatible with oxygenite tanks.

Any or all of these solutions, together with the upcoming anti-thievery improvements to outposts, would nerf O2 tanks without having to get rid of stacking or roll back the (apparently popular) recent buff to default O2 tank capacity.
